ls1intum/Artemis
# Refactor Gitlab User Management Service & Increase Coverage

### Checklist

### Motivation and Context


The current implementation of the `GitlabUserManagementService` class contains methods that are very long and difficult to understand. Furthermore the coverage is lower than 90%.

### Description

This PR breaks down big functions into many smaller ones which help the user understand what is happening. Tests are also added in order to increase the coverage > 90%.

This pr also fixes #3413 by changing the instructor's access level from `Maintainer` to `Owner` and the editor's access from `Developer` to `Maintainer`.

### Steps for Testing

This PR affects the user management of Gitlab so it should be tested on TS2.

The best way to test is to:
1. Create a instructor, tutor, and editor users through the user management view. 
2. Add all three users into a course that has a programming exercise
3. Log into Gitlab and locate the repositories of the exercise. Click on one of them 
4. On the left side, click on "Members" -> You should see the instructor, tutor and editor users
5. The instructor should have Maintainer role, editor the Developer, and tutor the Reporter roles
6. Now go back to the course a remove one or two users from the group
7. If you refresh Gitlab, they shouldn't be displayed anymore

### Test Coverage



GitLabUserManagementService.java 98%
